extensions/sockets/src/main/java/org/mule/extension/socket/api/provider/tcp/TcpRequesterProvider.java

```diff
@@ -55,29 +57,29 @@
    */
   @Parameter
   @Optional
-  @Placement(group = TLS_CONFIGURATION, tab = TLS)
+  @Placement(tab = TLS)
   @DisplayName(TLS_CONFIGURATION)
   private TlsContextFactory tlsContext;
 
   /**
    * This configuration parameter refers to the address where the {@link Socket} should connect to.
    */
-  @ParameterGroup
-  private ConnectionSettings connectionSettings;
+  @ParameterGroup(CONNECTION)
+  private SocketConnectionSettings connectionSettings;
 
   /**
    * {@link Socket} configuration properties
    */
-  @ParameterGroup
+  @ParameterGroup(SocketProperties.GROUP_NAME)
   private TcpClientSocketProperties tcpClientSocketProperties;
 
   /**
    * This configuration parameter refers to the address where the {@link Socket} should bind to.
    */
   @Parameter
   @Optional
-  @Placement(group = "Local Address Settings")
-  ConnectionSettings localAddressSettings = new ConnectionSettings();
+  @Placement(tab = "Local Address Settings")
+  SocketConnectionSettings localAddressSettings = new SocketConnectionSettings();
```

Comment:
org.mule.runtime.extension.api.exception.IllegalParameterModelDefinitionException: Class 'org.mule.extension.socket.api.SocketConnectionSettings' cannot be used with '@NullSafe' parameter since it contains non optional fields